core/src/sumeragi/main_loop.rs

```diff
@@ -354,13 +366,14 @@ impl Sumeragi {
         // NOTE: This sends "Block committed" event,
         // so it should be done AFTER public facing WSV update
         self.send_events(events);
-        self.update_topology(&block_signees, block_topology);
+
+        self.current_topology = new_topology;
```

Comment:
If now we recreate topology every time what role `current_topology` has?

Reply:
Storing the current topology so that we can refer to it? You could probably refactor to create the topology on the spot when you want to use it. I don't think that this is important for this PR though. We shouldn't delay getting these fixes to iroha users.